### PR TITLE
added Search module

### PR DESCRIPTION
**Pull request by mongedelonge**

### CODE REVIEW

File: IRCBot/modules/duckduckgo.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
+from template import template
+
+import logging
+
+import duckduckgo, google
+import requests, bs4
+
+class raw(template):
```
Comment (mrt-prodz) [inline IRCBot/modules/duckduckgo.py]: Thanks anticore! Could you please rename the class to search?

**Comment by mrt-prodz** [comment]
Since it searches for duckduckgo and then google if no items returned do you think we should name the module file search.py ? We could eventually add more search engines later.

**Comment by mongedelonge** [comment]
sounds good to me. i'll change it soon

**Comment by mongedelonge** [comment]
done!

File: IRCBot/modules/search.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
+from template import template
+
+import logging
+
+import duckduckgo, google
+import requests, bs4
+
+class raw(template):
```
Comment (mrt-prodz) [inline IRCBot/modules/search.py]: We just need to rename the class from **raw** to **search** and we should be good anticore!

**Comment by mongedelonge** [comment]
My bad, didn't even notice that, I thought you just meant the filename.

**Comment by mrt-prodz** [comment]
It's all good!

**Comment by mrt-prodz** [comment]
I made a couple changes (will push commit later) with indentation and return not being apparently properly indented.

I have another issue with the google module, by default it's using the current folder or $HOME. Thing is, the user I'm running this bot from on my server has no rights to write into the current folder, neither owns a home folder.

``` python
# Cookie jar. Stored at the user's home folder.
home_folder = os.getenv('HOME')
if not home_folder:
    home_folder = os.getenv('USERHOME')
    if not home_folder:
        home_folder = '.'   # Use the current folder on error.
cookie_jar = LWPCookieJar(os.path.join(home_folder, '.google-cookie'))
try:
    cookie_jar.load()
except Exception:
    pass
```

I need to look at the module if it's possible to set the path to /tmp or a specific path where the user running the bot is allowed to write.

(quick workaround would be to set user $HOME to /tmp )

**Pull request merged**